File: slack-poster/src/main/java/dev/number6/slackposter/SlackService.java
package dev.number6.slackposter;

import dev.number6.slackposter.port.SlackPort;
import dev.number6.slackposter.model.PresentableChannelSummary;
import com.amazonaws.services.lambda.runtime.LambdaLogger;

import javax.inject.Inject;

public class SlackService {

    private SlackPort slackPort;

    @Inject
    public SlackService(SlackPort slackPort) {
        this.slackPort = slackPort;
    }

    public void handleNewImage(ChannelSummaryImage image, LambdaLogger logger) {
        logger.log("handling image: " + image);

        if (image.hasFinalUpdate()) {
            slackPort.postMessageToChannel(new PresentableChannelSummary(image), logger);
        }
    }
}

File: slack-poster/src/main/java/dev/number6/slackposter/SlackService.kt
package dev.number6.slackposter

import com.amazonaws.services.lambda.runtime.LambdaLogger
import dev.number6.slackposter.model.PresentableChannelSummary
import dev.number6.slackposter.port.SlackPort
import javax.inject.Inject

class SlackService @Inject constructor(private val slackPort: SlackPort) {

    fun handleNewImage(image: ChannelSummaryImage, logger: LambdaLogger) {
        image.let {
            logger.log("handling image: $it")
            if (it.hasFinalUpdate()) {
                slackPort.postMessageToChannel(PresentableChannelSummary(it), logger)
            }
        }
    }
}